### PR TITLE
Added support for typescript

### PR DESCRIPTION
**Pull request by dalmia**

### CODE REVIEW

File: src/components/Items/Header.vue [hunk 1]
```diff
@@ -11,58 +10,38 @@
...
       // styling class for the skip button
       skipButtonClass:
-        'bg-primary hover:bg-primary-hover p-1 pl-4 pr-4 sm:p-2 sm:pl-10 sm:pr-10 lg:p-4 lg:pl-10 lg:pr-10 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none h-full'
-    })
+        "bg-primary hover:bg-primary-hover p-1 pl-4 pr-4 sm:p-2 sm:pl-10 sm:pr-10 lg:p-4 lg:pl-10 lg:pr-10 rounded-md shadow-xl disabled:opacity-50 disabled:pointer-events-none h-full" as string,
+      // main styling class for this component
+      containerClass:
+        "px-6 md:px-8 xl:px-12 flex w-full bg-white justify-end p-1 space-x-2 mt-2" as string,
+      // config for the title of skip button
+      skipButtonTitleConfig: {
```
Comment (pritamps) [inline src/components/Items/Header.vue:29]: Why are these lines in a typescript PR?

File: src/components/Items/Header.vue [hunk 2]
```diff
@@ -11,58 +10,38 @@
...
-      type: Boolean
-    },
-    isSkipEnabled: {
-      default: true,
-      type: Boolean
-    }
-  },
-  emits: ['skip-item']
-}
+  emits: ["skip-item"],
```
Comment (pritamps) [inline src/components/Items/Header.vue:45]: Why in TS PR?

File: src/components/UI/Buttons/IconButton.vue [hunk 3]
```diff
@@ -22,126 +22,137 @@
...
+      } as IconButtonTitleConfig,
+    });
+
+    // computed properties
     const innerContainerStyleClass = computed(() => {
       return [
         {
-          'flex-col': isStackedVertically.value,
-          'space-x-2': !isStackedVertically.value
+          "flex-col": isStackedVertically.value,
```
Comment (pritamps) [inline src/components/UI/Buttons/IconButton.vue:48]: Are these changes made by the TS linter?

File: src/components/UI/Buttons/IconButton.vue [hunk 4]
```diff
@@ -22,126 +22,137 @@
...
     // whether the icon and text need to be stacked vertically
     const isStackedVertically = computed(() => {
-      return props.orientation == 'vertical'
-    })
+      return props.orientation == "vertical";
+    });
+
+    // merges the default icon config and the icon config coming
+    // as a prop -> places that into "localIconConfig"
+    const localIconConfig = computed((): IconButtonIconConfig => {
```
Comment (pritamps) [inline src/components/UI/Buttons/IconButton.vue:61]: Why in TS PR?

File: src/types.ts [hunk 5]
```diff
@@ -0,0 +1,29 @@
+// contains all the types that we want to use
```
Comment (pritamps) [inline src/types.ts:1]: All types across the whole of the codebase?

**Review by pritamps — CHANGES_REQUESTED** [review]
Don't really have much idea of TS, so maybe I'm off base here, but looks like this PR has more than just TS ka support?

File: src/types.ts (hunk 5 above)
Comment (dalmia) [reply src/types.ts:1]: these are custom types - I'll update the comment.

**Pull request merged**